server/src/main/java/com/linecorp/centraldogma/server/auth/Session.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.server.auth;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.Serializable;
+import java.time.Duration;
+import java.time.Instant;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.centraldogma.internal.Util;
+
+/**
+ * An authenticated session which can be replicated to the other Central Dogma replicas as a serialized form.
+ */
+public final class Session implements Serializable {
```

Comment:
Could we:

- implement `Externalizable` for easier maintenance or
- serialize a `Session` as a JSON and only use serialized form for `rawSession`, e.g.

  ```json
  {
    "id": ...
    "username": ...
    "creationTime": "....",
    "expirationTime": "...",
    "rawSession": "<base64>"
  }
  ```

  .. by modifying `SessionJson(Deser|Ser)ializer`?